modDbImpl/src/org/aion/db/impl/redisdb/RedisClusterWrapper.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    @Override
+    public Set<byte[]> keys() {
+        check();
+
+        Set<byte[]> keys = new HashSet<>();
+        Map<String, JedisPool> clusterNodes = db.getClusterNodes();
+
+        for (String k : clusterNodes.keySet()) {
+            JedisPool jp = clusterNodes.get(k);
+            Jedis connection = jp.getResource();
+            try {
+                Set<String> keysAsString = connection.keys("*");
+
+                for (String key : keysAsString) {
+                    keys.add(key.getBytes(DBConstants.CHARSET));
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        return keys;
+    }
+
+    @Override
+    public void put(byte[] key, byte[] value) {
+        check(key);
+        check();
+
+        try {
+            db.set(new String(key, DBConstants.CHARSET), new String(value, DBConstants.CHARSET));
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    @Override
+    public void delete(byte[] key) {
+        check(key);
+        check();
+
+        try {
+            db.del(new String(key, DBConstants.CHARSET));
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void putBatch(Map<byte[], byte[]> inputMap) {
+        check(inputMap.keySet());
+        check();
+
+        try {
+            for (Map.Entry<byte[], byte[]> e : inputMap.entrySet()) {
+
+                byte[] key = e.getKey();
+                byte[] value = e.getValue();
+
+                if (value == null) {
+                    db.del(new String(key, DBConstants.CHARSET));
+                } else {
+                    db.set(new String(key, DBConstants.CHARSET), new String(value, DBConstants.CHARSET));
+                }
+            }
+
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void putToBatch(byte[] key, byte[] value) {
+        check(key);
+        check();
+
+        try {
+            if (value == null) {
+                db.del(new String(key, DBConstants.CHARSET));
+            } else {
+                db.set(new String(key, DBConstants.CHARSET), new String(value, DBConstants.CHARSET));
+            }
+
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void commitBatch() {
+        // not using batch here
+    }
+
+    @Override
+    public void deleteBatch(Collection<byte[]> keys) {
+        check(keys);
+
+        check();
+
+        try {
+            for (byte[] key : keys) {
+                db.del(new String(key, DBConstants.CHARSET));
+            }
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void close() {
+        try {
+            db.close();
```

Comment:
the `isOpen()` call would still return `true` after this call; perhaps the db object should be set to null and we can check for null in `isOpen()`